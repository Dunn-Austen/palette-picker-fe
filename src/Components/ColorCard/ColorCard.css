.color-card {
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  height: 100%;
  width: 200px;
}

/* Be aware of visibility concerns due to contrast with shifting background */
.hex-display {

}

.save-btn {
  background-color: beige;
  font-size: 20px;
  height: 30px;
  width: 70px;
}

.save-btn:hover {
  background-color: yellow;
  border: 2px solid black;
}
