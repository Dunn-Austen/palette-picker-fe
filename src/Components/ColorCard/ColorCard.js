import React from 'react';
import './ColorCard.css';
import PropTypes from 'prop-types';

const ColorCard = ({ hexCode, isSaved, setIsSaved }) => {

  const toggleSavedStatus = (isSaved) => {
    return !isSaved
  }

  return (
    <article className='color-card' style={{backgroundColor: hexCode}}>
      <h1 className='hex-display'>
        {hexCode}
      </h1>
      <button className='save-btn' onClick={() => setIsSaved(toggleSavedStatus(isSaved))}>
        {!isSaved ? 'Save' : 'Unlock'}
      </button>
    </article>
  );
}

export default ColorCard;

ColorCard.propTypes = {
  hexCode: PropTypes.string,
  isSaved: PropTypes.bool,
  setIsSaved: PropTypes.func
}
