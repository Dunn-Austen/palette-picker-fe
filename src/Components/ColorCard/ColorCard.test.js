import React from 'react';
import { shallow } from 'enzyme';
import ColorCard from './ColorCard';

describe('ColorCard', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<ColorCard />);

    expect(wrapper).toMatchSnapshot();
  });
})
