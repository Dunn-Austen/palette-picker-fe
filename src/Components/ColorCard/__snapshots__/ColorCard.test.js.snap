// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<article
  className="color-card"
  style={
    Object {
      "backgroundColor": undefined,
    }
  }
>
  <h1
    className="hex-display"
  />
  <button
    className="save-btn"
    onClick={[Function]}
  >
    Save
  </button>
</article>
`;
